src/coreclr/vm/readytorunstandalonemethodmetadata.cpp

```diff
@@ -0,0 +1,694 @@
+            case mdtTypeSpec: codedIndexType = 4; break;
+            default:
+                ThrowHR(COR_E_BADIMAGEFORMAT);
+        }
+
+        return rid + codedIndexType;
+    }
+
+    uint32_t GetAlternateToken(uint32_t inputToken)
+    {
+        uint32_t alternativeToken = 0;
+        if (!alternateTokens.Lookup(inputToken, &alternativeToken))
+        {
+            if ((TypeFromToken(inputToken) == mdtTypeDef) || (TypeFromToken(inputToken) == mdtTypeRef))
+            {
+                pTypeRefTokenStream->Append(inputToken);
+                alternativeToken = TokenFromRid((uint32_t)pTypeRefTokenStream->GetCount(), mdtTypeRef);
+            }
+            else
+            {
+                uint32_t newTokenType = 0;
+                SigBuilder blob;
+                switch (TypeFromToken(inputToken))
+                {
+                    case mdtString:
+                    {
+                        newTokenType = mdtString;
+                        LPCWSTR wszUserString;
+                        DWORD cchString;
+                        IfFailThrow(pMDImport->GetUserString(inputToken, &cchString, NULL, &wszUserString));
+                        blob.AppendData(cchString);
+                        blob.AppendBlob((void * const)wszUserString, sizeof(WCHAR) * cchString);
+                        // TODO: consider encoding via wtf-8, or possibly utf-8
+                        break;
+                    }
+                    case mdtTypeSpec:
+                    {
+                        newTokenType = mdtTypeSpec;
+                        PCCOR_SIGNATURE sigPtr;
+                        ULONG cbSig;
+                        IfFailThrow(pMDImport->GetTypeSpecFromToken(inputToken, &sigPtr, &cbSig));
+                        SigParser typeSpecSig(sigPtr, cbSig);
+                        StandaloneSigTranslator sigTranslator(&typeSpecSig, &blob, this);
+                        sigTranslator.ParseType();
+                        break;
+                    }
+                    case mdtMemberRef:
+                    {
+                        newTokenType = mdtMemberRef;
+                        PCCOR_SIGNATURE sig;
+                        ULONG cbSig;
+                        LPCSTR name;
+                        mdToken memberRefParent;
+
+                        IfFailThrow(pMDImport->GetNameAndSigOfMemberRef(inputToken, &sig, &cbSig, &name));
+                        IfFailThrow(pMDImport->GetParentOfMemberRef(inputToken, &memberRefParent));
+
+                        SigParser memberRefSigParse(sig, cbSig);
+                        
+                        StandaloneSigTranslator sigTranslator(&memberRefSigParse, &blob, this);
+                        sigTranslator.ParseMemberRefSignature();
+                        ULONG strLen = (ULONG)strlen(name); // Cast to ULONG is safe, as the data is held in a PE file
+                        blob.AppendData((ULONG)strlen(name));
```

Comment:
Nit - do we need to calculate <code>strlen</code> a second time considering we have evaluated it at line 294?